typescript/src/entities/draft.ts

```diff
@@ -48,4 +48,32 @@ export class Draft extends Entity {
 
   @Draft.property()
   public job?: Job;
+
+  public whereChangesRequested = () => {
+    /* true if any comment is/was a change request comment. */
+    if (this.comments === undefined) {
+      throw "comments is undefined. did you forget to embed it?";
+    }
+    for (const comment of this.comments) {
+      if (comment.changeRequest === undefined) {
```

Comment:
I suspect of this behavior.

I might add comment to draft with changesRequested not provided, if that is the case, I think whereChangesRequested should be false rather than throw a error.

also why the function name is "whereChangesRequested" in merchi.js the name is "changesRequested"

Reply:
> whereChangesRequested"

you've found a typo, it should be `were`. The reason is that it's a predicate.

Reply:
>  comment to draft with changesRequested not provided


yes. if information about whether changes where requested or not, was not provided, then, we cannot say whether or not changes where requested.

as a general principle for computer programming: if you don't know, don't guess -- it causes a very large amount of trouble when debugging (and it makes the bugs to start with).